### ISSUE TITLE
The order of concatenation for input'embedding and hidden_states in MTP is different from Deepseek-V3's paper.

### ISSUE THREAD

**Issue by tiggerwu** [sgl-project/sglang#6423]
Hello, I found the order of concatenation for input'embedding and hidden_states in MTP is different from the description in Deepseek-V3's paper. So I'm curious whether there are some considerations for this implementation in Sglang.

For Sglang, the order of concatenation is:
'''
torch.cat(
                (
                    self.enorm(hidden_states),
                    self.hnorm(forward_batch.spec_info.hidden_states),
                ),
                dim=-1,
            )
'''
![Image](https://github.com/user-attachments/assets/b7afaff6-6af9-4b1e-bc83-796128ced450)

For Deepseek-V3's paper, the order of concatenation is different:
![Image](https://github.com/user-attachments/assets/ec899e38-42a5-4fdf-9b32-6ef7c0b3c5e6)

**Comment by ispobock** [comment]
same issue ref: #3798

**Comment by tiggerwu** [comment]
Got it. Thanks!
